okhttp/src/main/java/okhttp3/internal/connection/ExchangeFinder.kt

```diff
@@ -59,7 +62,9 @@ class ExchangeFinder(
   // State guarded by connectionPool.
   private var routeSelector: RouteSelector? = null
   private var connectingConnection: RealConnection? = null
-  private var hasStreamFailure = false
+  private var refusedStreamCount = 0
```

Comment:
From [PR 5785](https://github.com/square/okhttp/pull/5785):
> What do you think about pulling these out into a separate class. Might help land it e.g. with a simple legacy strategy and the correct ones that you can unit test for?

Reply:
Love that idea. What I’d like to move towards is giving users the ability to define their own connection pooling policies including load balancing, seeding the connection pool, etc.